src/groovy/org/pih/warehouse/inventory/RefreshProductAvailabilityEvent.groovy

```diff
@@ -34,7 +34,8 @@ class RefreshProductAvailabilityEvent extends ApplicationEvent {
         this.locationId = source?.isInternalLocation() ? source?.parentLocation?.id : source?.id
         this.productIds = []
         this.forceRefresh = forceRefresh
-        this.disableRefresh = disableRefresh
+        // Avoid refreshing PA for all location, when there is no locationId
+        this.disableRefresh = this.locationId ? disableRefresh : Boolean.TRUE
```

Comment:
This should probably be the only commit for this ticket. The other issues should have separate bug tickets that include the stacktrace and/or steps to reproduce.